app/models/school.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class School < ApplicationRecord
+  include PgSearch::Model
+
+  pg_search_scope :search, against: %i[urn name town postcode], using: { tsearch: { prefix: true } }
```

Comment:
Do we need to add some gin/gist indexes on these fields? Would be good to test this with explain using the data from this PR https://github.com/DFE-Digital/register-trainee-teachers/pull/785

Reply:
@FriedSock I've added another migration to use a `tsvector` column and `gin` index to speed up search. There is a drawback however. In order to get the tests to pass, I had to change the active record schema format to `sql` otherwise the trigger will not be created for the test database since it gets updated using `rake db:schema:load` and unfortunately `schema.rb` doesn't support trigger configurations. I'm not happy about this, but is it a trade off worth accepting? cc: @AbigailMcP

Reply:
How about in the migration:

```
  def up
    add_column :schools, :searchable, :tsvector
    add_index :schools, :searchable, using: :gin

    execute <<-SQL
      CREATE TRIGGER tsvectorupdate BEFORE INSERT OR UPDATE
      ON schools FOR EACH ROW EXECUTE PROCEDURE
      tsvector_update_trigger(searchable, 'pg_catalog.english', urn, name, postcode, town);
    SQL

    School.find_each(&:touch)
  end
```

i.e. we `touch` all the schools to kick the trigger?

Reply:
`touch` won't work - it's only useful if we had existing records and we wanted to populate `searchable` after the column `searchable` gets added. If you view the `structure.sql` file you'll see the following line:

```sql
CREATE TRIGGER tsvectorupdate BEFORE INSERT OR UPDATE ON public.schools FOR EACH ROW EXECUTE FUNCTION tsvector_update_trigger('searchable', 'pg_catalog.english', 'urn', 'name', 'postcode', 'town');
```

Unfortunately `schema.rb` doesn't support more advanced Postgres features like this.

Reply:
Going to merge this PR without the `tsvector` column and gin index enhancement. We'll do the performance enhancement later on.

Reply:
Thanks @jhenderson - there's definitely a chance that we'll need to add this and swap to using `structure.sql`, but for now let's get the code in and wait until we discover we have a performance issue.